pypeit/edgetrace.py

```diff
@@ -4862,12 +4956,52 @@ def get_slits(self):
             specmin = specmin[indx]/binspec
             specmax = specmax[indx]/binspec
 
+        if self.maskdef_id is not None:
+            # Check for mismatched `maskdef_id` in the left and right edges
+            mkd_id_mismatch = self.maskdef_id[self.is_left] != self.maskdef_id[self.is_right]
+            if np.any(mkd_id_mismatch):
+                msgs.warn("Mismatched `maskdefId` in left and right traces for {}/{} slits. ".format(
+                          self.maskdef_id[self.is_left][mkd_id_mismatch].size, self.nslits) +
+                          "Choosing the left edge `maskdefId` if it is not -99, otherwise choosing right one")
+            _maskdef_id = self.maskdef_id[gpm & self.is_left]
+            # choose the value of `maskdef_id` that is not -99.
+            mkd_id_bad = _maskdef_id == -99
+            # this may not work if the corresponding right edge is also -99. Assuming this is not the case
+            _maskdef_id[mkd_id_bad] = self.maskdef_id[gpm & self.is_right][mkd_id_bad]
+            if np.any(_maskdef_id == -99):
+                msgs.warn("{} slits do not have `maskdefId` assigned.".format(_maskdef_id[_maskdef_id == -99].size) +
+                          "They will not be included in the design table")
+
+            # Store the matched slit-design and object information in a table.
+            self._fill_design_table(_maskdef_id, self.coeff_b, self.coeff_t, self.omodel_bspat, self.omodel_tspat)
+            self._fill_objects_table(_maskdef_id)
+            _merged_designtab = table.join(self.design, self.objects, keys=['SLITID'])
+            _merged_designtab.sort('TRACEID')
+        else:
+            _maskdef_id = None
+            _merged_designtab = None
+
+        if self.align_slit is not None:
+            if np.any(self.align_slit):
+                self.edge_msk[:,self.align_slit] = self.bitmask.turn_on(self.edge_msk[:,self.align_slit], 'BOXSLIT')
```

Comment:
Ah, here it is.  So, can we just use `edge_msk` instead of also having `align_slit`?

Reply:
Done!